sdk/security_keyvault/src/clients/keyvault_client.rs

```diff
@@ -75,7 +75,9 @@ impl KeyvaultClient {
         let dt = OffsetDateTime::now_utc();
         let time = date::to_rfc1123(&dt);
 
-        url.set_query(Some(API_VERSION_PARAM));
+        if url.query().is_none() || !url.query().unwrap().contains(API_VERSION_PARAM) {
+            url.set_query(Some(API_VERSION_PARAM));
+        }
```

Comment:
The current implementation _always_ overrides the `api-version`.

I believe the intent of this PR is to set the query param `api-version` if it doesn't already exist in the URL, but if it does, use that version.  I think this is more in line with what you're trying to accomplish:

```suggestion
        if !url
            .query_pairs()
            .into_iter()
            .any(|(key, _)| key == API_VERSION_PARAM_NAME)
        {
            url.set_query(Some(API_VERSION_PARAM))
        }
```

Reply:
Note, this suggestion assumes this change at the top of the file

Replace:
```rust
const API_VERSION_PARAM: &str = formatcp!("api-version={}", API_VERSION);
```

With this:
```rust
const API_VERSION_PARAM_NAME: &str = "api-version";
const API_VERSION_PARAM: &str = formatcp!("{}={}", API_VERSION_PARAM_NAME, API_VERSION);
```

Reply:
@heaths , I believe this suggestion would address the concerns you highlighted in your comment.  Thoughts?

Reply:
Our REST API guidelines say to override the api-version with whatever the client was configured unconditionally.